src/services/fileParser.ts

```diff
@@ -0,0 +1,96 @@
+import { exec } from 'child_process';
+import { promisify } from 'util';
+import minimatch from 'minimatch';
+import { workspace } from 'vscode';
+import { FileParsingRule, getConfiguration } from './configuration';
+import { log } from './logger';
+
+const execAsync = promisify(exec);
+
+export interface ParsedFileResult {
+  originalPath: string;
+  parsedContent: string;
+}
+
+function getParsingRules(): FileParsingRule[] {
+  return getConfiguration('fileParsingRules') || [];
+}
+
+export function shouldParseFile(filePath: string): boolean {
+  const parsingRules = getParsingRules();
+  return parsingRules.some(rule =>
+    minimatch(filePath, rule.pattern, { matchBase: true })
+  );
+}
+
+function getParsingRule(filePath: string): FileParsingRule | undefined {
+  const parsingRules = getParsingRules();
+  return parsingRules.find(rule =>
+    minimatch(filePath, rule.pattern, { matchBase: true })
+  );
+}
+
+export async function parseFile(filePath: string): Promise<ParsedFileResult> {
+  const rule = getParsingRule(filePath);
+  if (!rule) {
+    throw new Error(`No parsing rule found for file: ${filePath}`);
+  }
+
+  try {
+    const parsedContent = await executeParsingCommand(filePath, rule);
+    const result: ParsedFileResult = {
+      originalPath: filePath,
+      parsedContent
+    };
+
+    return result;
+  } catch (error) {
+    const errorMessage = error instanceof Error ? error.message : String(error);
+    log(`Error parsing file ${filePath}: ${errorMessage}`);
+    throw new Error(`Failed to parse file ${filePath}: ${errorMessage}`);
+  }
+}
+
+async function executeParsingCommand(filePath: string, rule: FileParsingRule): Promise<string> {
+  const args = rule.args.map(arg =>
+    arg.replace('{file}', filePath)
+  );
+
+  const command = `${rule.command} ${args.join(' ')}`;
+  const workingDirectory = rule.workingDirectory || process.cwd();
+  const timeout = rule.timeout || 30000;
```

Comment:
set default in _package.json_

```suggestion
  const timeout = rule.timeout;
```